trunk/NDHMS/Data_Rec/module_RT_data.f90

```diff
@@ -0,0 +1,274 @@
+  CHARACTER(len=15), allocatable, DIMENSION(:) :: gages
+  !                                                    123456789012345
+  CHARACTER(len=15)                     :: gageMiss = '               '
+!  CHARACTER(len=15)                     :: gageMiss = '          -9999'
+
+  INTEGER, allocatable, DIMENSION(:)   :: TYPEL       !type of link Muskingum: 0 strm 1 lake
+                                                      !-- Diffusion: 0 edge or pour; 1 interior; 2 lake
+  INTEGER, allocatable, DIMENSION(:)   :: TYPEN      !type of link 0 strm 1 lake
+  REAL, allocatable, DIMENSION(:)      :: QLAKEI      !lake inflow in difussion scheme
+  REAL, allocatable, DIMENSION(:)      :: QLAKEO      !lake outflow in difussion scheme
+  INTEGER, allocatable, DIMENSION(:)   :: LAKENODE   !which nodes flow into which lakes
+  INTEGER, allocatable, DIMENSION(:)   :: LINKID     ! id of links on linked routing
+  REAL, allocatable, DIMENSION(:)      :: CVOL       ! channel volume
+  INTEGER, allocatable, DIMENSION(:,:)   :: pnode    !parent nodes : start from 2
+  integer :: maxv_p              ! array size for  second column of the pnode
+
+  REAL, allocatable, DIMENSION(:)      :: MUSK, MUSX !muskingum params
+  REAL, allocatable, DIMENSION(:)      :: CHANLEN    !link length
+  REAL, allocatable, DIMENSION(:)      :: MannN      !mannings N
+  REAL, allocatable, DIMENSION(:)      :: So     !link slope and channel conductivity (m/s)
+  REAL, allocatable, DIMENSION(:)      :: ChSSlp ! channel side slope
+  REAL, allocatable, DIMENSION(:)      :: Bw     ! Bottom width of channel
+  REAL, allocatable, DIMENSION(:)      :: Tw     ! top width of channel
+  REAL, allocatable, DIMENSION(:)      :: Tw_CC  ! top width of the (CC) compound channel
+  REAL, allocatable, DIMENSION(:)      :: n_CC   ! mannings N of (CC) compound channel
+
+  REAL, allocatable, DIMENSION(:,:)    :: QLINK      !flow in link
+  integer, allocatable, DIMENSION(:)   :: ascendIndex !sorting of routelink
+  REAL, allocatable, DIMENSION(:)      :: HLINK      !head in link
+  REAL, allocatable, DIMENSION(:)      :: ZELEV      !elevation of nodes for channel
+  INTEGER, allocatable, DIMENSION(:)   :: CHANXI,CHANYJ !map chan to fine grid
+  REAL,  DIMENSION(50)     :: BOTWID,TOPWID,HLINK_INIT,CHAN_SS,CHMann !Channel parms from table
+  REAL,  DIMENSION(50)     :: TOPWIDCC, NCC  !topwidth and mannings n of compund
+
+  ! VARIABLES FOR  RESERVOIRS
+  REAL, allocatable, DIMENSION(:)      :: RESHT  !reservoir height
+!-- lake params
+  INTEGER, allocatable, DIMENSION(:) :: LAKEIDA     !id of lakes in routlink file
+  INTEGER, allocatable, DIMENSION(:) :: LAKEIDM     !id of LAKES Modeled in LAKEPARM.nc or tbl
+  REAL, allocatable, DIMENSION(:)    :: HRZAREA    !horizontal extent of lake, km^2
+  REAL, allocatable, DIMENSION(:)    :: WEIRL      !overtop weir length (m)
+  REAL, allocatable, DIMENSION(:)    :: DAML      !overtop dam length (m)
```

Comment:
If we are adding NEW variables that don't require a significant refactor throughout the code, can we try to move away from fixed format, F77 naming conventions?  DAM_LENGTH is much more descriptive, especially when seeing it in the code out of context with the associated comment.  New variables are the easiest to start with! And it looks like the addition of dam length in module_levelpool did this, so this name should be the same!

Reply:
DAM_LENGTH=WEIRL ?
can I understand like that?

Reply:
@junjie2008v I'm not sure I understand what your question is asking.  Is it related to this code review comment or just looking for clarity on the `DAM_LENGTH` variable in general?   If this is the case, I suggest opening an issue to start that conversation.   IIRC there might be a closed issue related to `DAM_LENGTH` that might be a good place to start as well.

Reply:
Thank you for taking the time out of your busy schedule to reply！&& What suggestions do you have for the DAM_LENGTH ?